extensions/cornerstone/src/commandsModule.ts

```diff
@@ -415,6 +422,122 @@ const commandsModule = ({ servicesManager }) => {
         (activeViewportIndex - 1 + viewports.length) % viewports.length;
       ViewportGridService.setActiveViewportIndex(nextViewportIndex);
     },
+    activateMPRForActiveViewport: () => {
+      const { activeViewportIndex, viewports } = ViewportGridService.getState();
+      const viewportDisplaySetInstanceUIDs =
+        viewports[activeViewportIndex].displaySetInstanceUIDs;
+
+      const displaySetsToHang = viewportDisplaySetInstanceUIDs.map(
+        displaySetInstanceUID => {
+          const displaySet = DisplaySetService.getDisplaySetByUID(
+            displaySetInstanceUID
+          );
+
+          return displaySet;
+        }
+      );
+
+      if (displaySetsToHang.some(ds => !ds.isReconstructable)) {
+        UINotificationService.show({
+          title: 'Multiplanar reconstruction (MPR) ',
+          message:
+            'Cannot create MPR for this series since it is not reconstructable.',
+          type: 'warning',
+          displayTime: 3000,
+        });
+
+        return;
+      }
+
+      const matchingDisplaySets = {};
+
+      displaySetsToHang.forEach(displaySet => {
+        const {
+          displaySetInstanceUID,
+          SeriesInstanceUID,
+          StudyInstanceUID,
+        } = displaySet;
+
+        matchingDisplaySets[displaySetInstanceUID] = {
+          displaySetInstanceUID,
+          SeriesInstanceUID,
+          StudyInstanceUID,
+        } as Types.HangingProtocol.DisplaySetMatchDetails;
+      });
+
+      const hpViewports: Types.HangingProtocol.Viewport[] = [
+        'axial',
```

Comment:
Can you add acquisition here?

Reply:
I see what you are doing, you are generating a hanging protocol to apply.  Is there any reason not to allow applying a generic hanging protocol, and just define it in the module in the normal way?  That way one could define buttons/commands to apply arbitrary hanging protocols, and you wouldn't have this large chunk of data generation in the middle of the commands module.